### PR TITLE
Bug fix on accordion active item

### PR DESCRIPTION
**Pull request by nadijagraca**
## Description
Fixing bug on accordion active item:

What to test:
If relevant accordion item opens when navigating from homepage or by typing path in browser. 

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
  - I have not referenced individuals, products or companies in any commits, directly or indirectly.
  - I have not added data or restricted code in any commits, directly or indirectly.

### CODE REVIEW

**Review by huong-li-nguyen — APPROVED** [review]
Works well - thanks for fixing! 👍 

@nadijagraca - for the next time it would be great if you could add some example code snippet to see that the relevant changes work. I wasn't quite sure what I had to test first 😄 

For the next reviewer - replace the navigation in the example demo with:

```
navigation=vm.Navigation(
        nav_selector=vm.NavBar(
            items=[
                vm.NavLink(label="Homepage", pages=["Homepage"], icon="Home"),
                vm.NavLink(
                    label="Analysis",
                    pages={"Title 1": ["Variable Analysis", "Benchmark Analysis"],
                           "Title 2": ["Continent Summary", "Relationship Analysis"]},
                    icon="Stacked Bar Chart",
                ),
            ]
        ),
    ),
```

Then navigate to relationship-analysis either via clicking on the card on the homepage or by directly changing the path in the URL. What you should see is that the accordion opens accordingly (opens where active page is)

@nadijagraca @l0uden  - will we add screenshots to test this? or do the unit tests cover this use case sufficiently?

File: vizro-core/changelog.d/20240123_103311_nadija_ratkusic_graca_accordion_active_item.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+<!--
+### Deprecated
+
+- A bullet item for the Deprecated category with a link to the relevant PR at the end of your entry, e.g. Enable feature XXX ([#1](https://github.com/mckinsey/vizro/pull/1))
+
+-->
+
+### Fixed
+
+- Fixed bug on accordion, ensuring the relevant accordion section is opened when navigating from nav cards or by typing the path directly in browser. ([#276](https://github.com/mckinsey/vizro/pull/276))
```
Comment (huong-li-nguyen) [inline vizro-core/changelog.d/20240123_103311_nadija_ratkusic_graca_accordion_active_item.md:40]:
```suggestion
- Fix bug on accordion not opening properly when re-directing to active page via navigation card or URL path. ([#276](https://github.com/mckinsey/vizro/pull/276))
```

**Review by maxschulz-COL — APPROVED** [review]

File: vizro-core/src/vizro/models/_navigation/accordion.py [hunk 2]
```diff
@@ -53,9 +53,14 @@ def build(self, *, active_page_id=None):
                     children=accordion_buttons,
                     title=page_group.upper(),
                     class_name="accordion-item-header",
+                    item_id=page_group,
                 )
             )
 
+        active_item = next(
+            (page_group for page_group, page_members in self.pages.items() if active_page_id in page_members)
+        )
```
Comment (antonymilne) [inline vizro-core/src/vizro/models/_navigation/accordion.py:62]:
```suggestion
        active_item = next(
            page_group for page_group, page_members in self.pages.items() if active_page_id in page_members
        )
```

**Review by antonymilne — COMMENTED** [review]
Really nice fix! I didn't test it out but it looks great, thank you for doing it 👍 

However, I think there _might_ be a minor bug in it. In this code:

```
next(
    page_group for page_group, page_members in self.pages.items() if active_page_id in page_members
)
```
if `active_page_id in page_members` is always `False` then this will turn into `next` of an empty dictionary, which will raise `StopIteration` and crash. Try this out:
```
next(x for x, y in dict(a=1).items() if x == "b")
```

What I'm not sure about is whether this case can ever occur in practice? If I type the URL to a page that doesn't exist in the navigation then is it possible it would trigger this error?

File: vizro-core/src/vizro/models/_navigation/accordion.py (hunk 2 above)
Comment (nadijagraca) [reply vizro-core/src/vizro/models/_navigation/accordion.py]: Implemented, and it does not crash when navigating on non-existent page. Good catch. 🙌

**Pull request merged**